sdk/ml/azure-ai-ml/azure/ai/ml/_utils/utils.py

```diff
@@ -924,3 +924,77 @@ def write_to_shared_file(file_path: Union[str, PathLike], content: str):
             os.chmod(file_path, share_mode)
         except PermissionError:
             pass
+
+
+def _get_valid_dot_keys_with_wildcard_impl(
```

Comment:
Is it possible to use regex or something else for this? not sure if we need to maintain the matching function by ourselves.

Reply:
Hmm actually only ".*" or ".*." is supported in this function, so its implementation is not so complicated. I keep it here for 4 reasons:
1) This function is targeting to get dot_keys, for which the most widely used package is pydash, while there is neither an util function to do what we want directly, nor an util function to extract all valid dot keys, with which we can use common matching tool like regex for further processing.
2) With this function, we can make the attribute overwriting logic much easier to understand.
3) As I have mentioned above, we may control the effort to maintain the matching function by limit its functionality.
4) Such a function can be helpful to simplify pipeline job tests. For example, we frequently skip some fields for all steps in a pipeline job; previously we tend to manually list all steps.